Libraries/LibWeb/SVG/SVGGraphicsElement.cpp

```diff
@@ -285,6 +286,40 @@ float SVGGraphicsElement::resolve_relative_to_viewport_size(CSS::LengthPercentag
     return length_percentage.to_px(*layout_node(), scaled_viewport_size).to_double();
 }
 
+Vector<float> SVGGraphicsElement::stroke_dasharray() const
+{
+    if (!layout_node())
+        return {};
+
+    Vector<float> dasharray;
+    for (auto const& value : layout_node()->computed_values().stroke_dasharray()) {
+        value.visit(
+            [&](CSS::LengthPercentage const& length_percentage) {
+                dasharray.append(resolve_relative_to_viewport_size(length_percentage));
+            },
+            [&](CSS::NumberOrCalculated const& number_or_calculated) {
+                dasharray.append(number_or_calculated.resolved(*layout_node()));
+            });
+    }
+
+    // If the list has an odd number of values, then it is repeated to yield an even number of values.
+    if (dasharray.size() % 2 == 1)
+        dasharray.extend(dasharray);
+
+    // If any value in the list is negative, the <dasharray> value is invalid. If all of the values in the list are zero, then the stroke is rendered as a solid line without any dashing.
+    bool all_zero = true;
+    for (auto& value : dasharray) {
+        if (value < 0)
+            return {};
+        if (value != 0)
+            all_zero = false;
+    }
+    if (all_zero)
+        return {};
```

Comment:
Nitpick: this entire block of code could run _before_ potentially doubling the size of `dasharray`

Reply:
This is the order in the spec, and in practice it won't matter so I figured I'd match spec order. I can change it if you still want that, ofc; let me know what you prefer.

Reply:
Keeping the same order as in the spec sounds fine, but could you add a spec link?

Reply:
Done!